watchers/configmap.go

```diff
@@ -56,19 +58,23 @@ func (w *ConfigMapWatcher) Start(ctx context.Context) {
 }
 
 // watchChange consumes watch events and returns on change.
-func (w *ConfigMapWatcher) watchChange(ctx context.Context) error {
+func (w *ConfigMapWatcher) watchChange(ctx context.Context, waitDuration time.Duration) error {
 	watcher, err := w.client.Watch(ctx, w.listOpt)
 	if err != nil {
 		return err
 	}
 	defer watcher.Stop()
 
-	time.Sleep(time.Minute)
+	if waitDuration != 0 {
```

Comment:
I think the types don't match for this comparison

Reply:
```
package main

import (
	"fmt"
	"time"
)

func main() {
	fmt.Println(time.Second * time.Duration(0))
	fmt.Println(time.Second * time.Duration(0) == 0)
}
```
```
0s
true

Program exited.
```

Reply:
Yep, I was silly and thinking about `time.Time`